testing/src/main/java/io/stargate/it/grpc/SchemaChangesTest.java

```diff
@@ -0,0 +1,90 @@
+package io.stargate.it.grpc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.google.protobuf.Empty;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import io.stargate.grpc.StargateBearerToken;
+import io.stargate.it.BaseIntegrationTest;
+import io.stargate.it.driver.CqlSessionExtension;
+import io.stargate.it.driver.TestKeyspace;
+import io.stargate.it.http.RestUtils;
+import io.stargate.it.storage.StargateConnectionInfo;
+import io.stargate.proto.QueryOuterClass.SchemaChange;
+import io.stargate.proto.StargateGrpc;
+import java.io.IOException;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(CqlSessionExtension.class)
+public class SchemaChangesTest extends BaseIntegrationTest {
+
+  private StargateGrpc.StargateStub asyncStub;
+
+  @BeforeEach
+  public void setup(StargateConnectionInfo cluster) throws IOException {
+    String seedAddress = cluster.seedAddress();
+    ManagedChannel channel =
+        ManagedChannelBuilder.forAddress(seedAddress, 8090).usePlaintext().build();
+    asyncStub =
+        StargateGrpc.newStub(channel)
+            .withCallCredentials(new StargateBearerToken(RestUtils.getAuthToken(seedAddress)));
+  }
+
+  @Test
+  @DisplayName("Should receive table changes")
+  public void tableChangesTest(CqlSession session, @TestKeyspace CqlIdentifier keyspace) {
+    SchemaChangeAccumulator changes = new SchemaChangeAccumulator();
+    asyncStub.getSchemaChanges(Empty.newBuilder().build(), changes);
+
+    // TODO why does this not generate an event?
+    session.execute("CREATE TABLE foo(k int PRIMARY KEY)");
```

Comment:
This is weird. I've tried with UDTs as well, and same thing, CREATE does not generate a notification but DROP does.

Reply:
because the listener is not yet registered once you do the `execute`, you call the async api with the GRPC and return thus immediately, it's not guaranteed that the execute will be done after the `persistence.registerEventListener(this);` is called in the `SchemaChangesHandler`

Reply:
Good catch! Pausing after line 46 does indeed fix the issue.